### ISSUE TITLE
Adjust content metadata determination to be based on rights 

### ISSUE THREAD

**Issue by peetucket** [sul-dlss/sdr-api#90]
See also #13, which may be closed in favor of this ticket.

A discussion on 1/17/2020 led to the following issues:

1. If we shelve everything, we will run out of disk space on the stacks.
2. We thus should only shelve content this is actually world accessible, ie. content that meets the rights definition of public domain as computed in https://github.com/sul-dlss/google-books/issues/1
3. Content that is is within 5 years of being in the public domain should also be shelved, but made citation only with an embargo (also no change from what is noted in https://github.com/sul-dlss/google-books/issues/1) 
4. However, content that is greater than 5 years of being in the public domain will be set to citation only, and shelve=no/publish=no for all files.  It will still have an embargo as defined in https://github.com/sul-dlss/google-books/issues/1.  But this will result in far less disk space used on the stacks.

So basically contentMetadata generation will get more complicated, and require the ability to determine the copyright status in order to decide on the default publish/preserve/shelve attributes.

**Comment by jcoyne** [comment]
@peetucket If we don't mark an item as shelve, then will the repository administrators of the future need to come back and fix all the objects?  Will they know that "shelve=false" means (until the copyright period expires?)  Instead should we change the shelve robot to not shelve anything that is not world readable?

**Comment by jcoyne** [comment]
@peetucket why would we do "publish=no" doesn't that mean it isn't even citation-only?

**Comment by peetucket** [comment]
since publish=no is at the file level, you will still get a "citation only" PURL as long as the object rights declare that it is world discoverable...it just means that no files are there to be had on the PURL (dark is if set discoverable to none in the object rights, in which case you don't get a PURL at all)

**Comment by jcoyne** [comment]
So, what is the difference on PURL if publish is yes or no?

**Comment by peetucket** [comment]
> So, the difference would be that this code https://github.com/sul-dlss/dor-services-app/blob/e2a654f55c34909ffe0d7524d0c1994d4a98dc05/app/services/public_xml_service.rb#L83 would make it so that we don't publically acknowledge that we have a file. 


Yes, I believe it just influences what goes into the public XML sent to PURL, which is what the embed player uses to build the UI (either linking to files, or showing images/media/etc.)

>Is there ever a case where one would want publish and shelve to not have matching values?

That's the question I asked in the slack channel a few days ago, and the answer seemed to boil down to possibly web-accessioning only, which needs files on the stacks, but not detected by the embed player (since the web accessioning "player" is the way back machine).
